s3pypi/__main__.py

```diff
@@ -3,77 +3,50 @@
-    p.add_argument("--secret", help="S3 secret")
-    p.add_argument("--region", help="AWS region")
-    p.add_argument("--profile", help="AWS profile")
-    p.add_argument("--force", action="store_true", help="Overwrite existing packages")
-    p.add_argument(
-        "--no-wheel", dest="wheel", action="store_false", help="Skip wheel distribution"
-    )
     p.add_argument(
-        "--no-sdist", dest="sdist", action="store_false", help="Skip sdist distribution"
+        "dist",
+        nargs="+",
+        type=Path,
+        help="The distribution files to upload to S3. Usually `dist/*`.",
     )
+    p.add_argument("-b", "--bucket", required=True, help="The S3 bucket to upload to.")
+    p.add_argument("--profile", help="Optional AWS profile to use.")
+    p.add_argument("--region", help="Optional AWS region to target.")
+    p.add_argument("--prefix", help="Optional prefix to use for S3 object names.")
+    p.add_argument("--acl", help="Optional canned ACL to use for S3 objects.")
     p.add_argument(
-        "--dist-path", help="Path to directory with wheel/sdist to be uploaded"
+        "--unsafe-s3-website",
+        action="store_true",
+        help=(
+            "Store the index as an S3 object named `<package>/index.html` instead of `<package>/`. "
+            "This option is provided for backwards compatibility with S3 website endpoints, "
+            "the use of which is discouraged because they require the bucket to be publicly accessible. "
+            "It's recommended to instead use a private S3 bucket with a CloudFront Origin Access Identity."
+        ),
     )
-    p.add_argument(
-        "--bare", action="store_true", help="Store index as bare package name"
-    )
-
-    acl_group = p.add_mutually_exclusive_group()
-
-    acl_group.add_argument(
-        "--private",
-        action="store_const",
-        const="private",
-        default="public-read",
-        dest="acl",
-        help="Store S3 Keys as private objects",
-    )
-    acl_group.add_argument(
-        "--acl", default="public-read", help="ACL to use for S3 objects"
-    )
-
-    p.add_argument("--verbose", action="store_true", help="Turn on verbose output.")
+    p.add_argument("--force", action="store_true", help="Overwrite existing packages.")
+    p.add_argument("--verbose", action="store_true", help="Enable verbose output.")
     p.add_argument("--version", action="version", version=__version__)
-
     return p.parse_args(args)
 
 
 def main(*args):
-    args = parse_args(args or sys.argv[1:])
-
-    log.setLevel(logging.DEBUG if args.verbose else logging.INFO)
+    kwargs = vars(parse_args(args or sys.argv[1:]))
+    log.setLevel(logging.DEBUG if kwargs.pop("verbose") else logging.INFO)
 
     try:
-        create_and_upload_package(args)
-    except S3PyPiError as e:
+        core.upload_packages(**kwargs)
+    except core.S3PyPiError as e:
         print("error: %s" % e)
         sys.exit(1)
```

Comment:
These 2 lines are equivalent to:
```suggestion
        sys.exit(f"error: {e}")
```

From [sys.exit](https://docs.python.org/3.8/library/sys.html#sys.exit):

>  If another type of object is passed, None is equivalent to passing zero, and any other object is printed to stderr and results in an exit code of 1. In particular, sys.exit("some error message") is a quick way to exit a program when an error occurs.

Of course 'explicit is better than implicit', so I'd understand if you prefer this!